src/collector.ts

```diff
@@ -17,73 +17,99 @@ enum ValueType {
 };
 
 /* Value variant */
-interface IVariant {
+export interface IVariant {
   type:   ValueType;
   object: any;
 }
 
 /* Line and column inside the JSON file */
-interface IPosition {
+export interface IPosition {
   line:   number;
   column: number;
 };
 
 /* Key/Value entry with positions */
-interface IKeyValueEntry {
+export interface IKeyValueEntry {
   key:            string;
   value:          IVariant;
   key_position:   IPosition;
   value_position: IPosition;
 };
 
-class KeyValueEntry implements IKeyValueEntry {
+export class KeyValueEntry implements IKeyValueEntry {
   key:            string;
   value:          IVariant;
   key_position:   IPosition;
   value_position: IPosition;
 
-  constructor(k: string, pos: IPosition) {
+  constructor(k: string, pos: IPosition, v?: IVariant, v_pos?: IPosition) {
     this.key = k;
     this.key_position = pos;
+    this.value = v;
+    this.value_position = v_pos;
   }
 }
 
-class Variant implements IVariant {
-    constructor(public type: ValueType, public object: any) {}
+export class Variant implements IVariant {
+  constructor(public type: ValueType, public object: any) {}
 }
 
 /* String value with position */
-interface IPositionedString {
+export interface IPositionedString {
   value:    string;
   position: IPosition;
 }
 
 /* Dependency specification */
-interface IDependency {
-  name:    IPositionedString;
+export interface IDependency {
+  name: IPositionedString;
   version: IPositionedString;
 }
 
+export interface IHashableDependency extends IDependency {
+  key(): string;
+}
+
 /* Dependency collector interface */
-interface IDependencyCollector {
+export interface IDependencyCollector {
   classes: Array<string>;
   collect(contents: string): Promise<Array<IDependency>>;
 }
 
 /* Dependency class that can be created from `IKeyValueEntry` */
-class Dependency implements IDependency {
+export class Dependency implements IHashableDependency {
   name:    IPositionedString;
   version: IPositionedString;
   constructor(dependency: IKeyValueEntry) {
     this.name = {
-        value: dependency.key,
-        position: dependency.key_position
+      value: dependency.key,
+      position: dependency.key_position
     };
     this.version = {
-        value: dependency.value.object,
-        position: dependency.value_position
+      value: dependency.value.object,
+      position: dependency.value_position
     };
   }
+
+  key(): string {
+    return `${this.name.value}@${this.version.value}`;
```

Comment:
In case of golang '@' separator is used to separate package and module.Can we use some other separator between name and version?

Reply:
key() just generates an opaque string which will be a candidate key for HashMap. It doesn't have to have any meaning.